spread.yaml

```diff
@@ -270,6 +270,38 @@ suites:
                 apt-get purge -y snapd
             fi
 
+    tests/completion/:
+        summary: completion tests
+
+        # ppc64el disabled because of https://github.com/snapcore/snapd/issues/2502
+        systems: [-ubuntu-core-*, -ubuntu-*-ppc64el]
+
+        prepare: |
+            . $TESTSLIB/prepare.sh
+            prepare_classic
+        prepare-each: |
+            $TESTSLIB/reset.sh --reuse-core
+            . $TESTSLIB/prepare.sh
+            prepare_each_classic
+        restore: |
+            $TESTSLIB/reset.sh --store
+            apt-get purge -y snapd
+
+        environment:
```

Comment:
Each completion test tries to complete something by double-tabbing on empty, double-tabbing at the start of a word, single-tabbing something that returns a single match, and double-tabbing something that returns some but not all matches. This same thing is run "directly" (from `tests/completion/simple` and "indirectly" via snap completion (from `tests/completion/indirect`). These directories set up the tests, but the tests themselves key off of the variant to source the appropriate `.vars` (which sets what to complete, and what to expect of that completion), `.sh` (for test-specific setup; e.g. changing to a particular directory to test particular filename comletions), and `.completion` (the completion snippet itself).

HTH?